securedrop_client/gui/widgets.py

```diff
@@ -839,33 +839,31 @@ def update(self, sources: List[Source]):
         """
         Reset and update the list with the passed in list of sources.
         """
-        # A flag to show if a source is currently selected (the current source
-        # doesn't have to be selected in a QListWidget -- it appears to default
-        # to whatever is in row 0, whether it's selected or not).
-        has_source_selected = False
-        if self.currentRow() > -1:
-            has_source_selected = self.item(self.currentRow()).isSelected()
-        current_source = self.get_current_source()
-        current_source_id = current_source and current_source.id
```

Comment:
confirming: since we no longer need to redraw the whole view, we don't need this logic to preserve the current source selection